### ISSUE TITLE
[v3] Animated.Text emits warning for array styles

### ISSUE THREAD

**Issue by jeremybarbet** [jpudysz/react-native-unistyles#512]
### Description

Hi there,

If I have the following code

index.tsx
```tsx
const Index = () => (
    <TextComponent name="Some text" style={s.text} />
);

const s = StyleSheet.create(theme => ({
  text: {
    color: theme.primary,
  },
}));
```

TextComponent.tsx
```tsx
export const TextComponent = ({ name, style }: TextComponentProps) => (
  <Animated.Text style={[s.wrapper, style]}>{name}</Animated.Text>
);

const s = StyleSheet.create(theme => ({
  wrapper: {
    color: theme.primary,
  },
}));
```

it will emits `Unistyles: we detected style object with 2 unistyles styles. This might cause no updates or unpredictable behavior. Please check style prop for "Text" and use array syntax instead of object syntax. [Component Stack]`

If changing to

```diff
export const TextComponent = ({ name, style }: TextComponentProps) => (
-  <Animated.Text style={[s.wrapper, style]}>{name}</Animated.Text>
+  <Text style={[s.wrapper, style]}>{name}</Text>
);
```

It doesn't emit any warnings anymore, but obvisously reanimated doesn't work anymore

### Steps to Reproduce

1. Follow this PR https://github.com/jeremybarbet/unistyles-sandbox/pull/1/files
2. You can also clone the repository `git@github.com:jeremybarbet/unistyles-sandbox.git` and run the app on the `repro/animated.text` branch

### Snack or Repository Link (Optional)

https://github.com/jeremybarbet/unistyles-sandbox

### Unistyles Version

3.0.0-nightly-20250121

### React Native Version

0.76.6

### Platforms

iOS

### Expo

Yes

**Comment by jpudysz** [comment]
This is because of `Reanimated`. They flatten the array inside `createAnimatedComponent`. It's before I have a chance to process it. 

I will think about potential solution (if any) and update this thread. For now moving to `beta.7`.

**Comment by jpudysz** [comment]
Ok, I will create a PR to support that in Reanimated. Moved to next beta as it might take few days/weeks.

**Comment by jpudysz** [comment]
For ref: https://github.com/software-mansion/react-native-reanimated/pull/7021

**Comment by Brentlok** [comment]
patch:

[react-native-reanimated+3.16.7.patch](https://github.com/user-attachments/files/18781539/react-native-reanimated%2B3.16.7.patch)

**Comment by jpudysz** [comment]
I would prefer to leave a note in the docs. 

It works perfectly fine with Rea 3 and we just need to update one line , so I wouldn't block it for users.

**Comment by jpudysz** [comment]
Left a note in the docs. There is nothing more I can do about it. 

Anyone that wants to get this going with REA 3.0, please grab patch from above - you just need to change one line. 


With REA 4.0-beta.3 it will work out of the box.

**Comment by csheppard** [comment]
This warning is back for reanimated 4.3.0 using Animated.View.

Working fine with 4.2.3.

**Comment by DragomirAndrei19** [comment]
I can confirm, my application is now full of these errors.

It's quite unfortunate as we've just completed the React Native upgrade 0.84.1 to 0.85.1.

We can't go back to reanimated 4.2.3 as it doesn't support React Native 0.85...

**Comment by Arthi-chaud** [comment]
@csheppard @DragomirAndrei19 Reanimated just released [v4.3.1](https://github.com/software-mansion/react-native-reanimated/releases) which fixes the regression! (https://github.com/software-mansion/react-native-reanimated/pull/9284)
